### PR TITLE
Apease Clippy and some renames

### PR DESCRIPTION
**Pull request by dvdplm**

### CODE REVIEW

File: types/src/v2/traits.rs [hunk 1]
```diff
@@ -11,5 +11,5 @@ impl<R, T> RpcMethod<R> for T where T: Fn(RpcParams) -> Result<R, RpcError> + Se
 
 /// RPC Call Result.
 trait RpcResult {
-	fn to_json(self, id: Option<&RawValue>) -> anyhow::Result<String>;
+	fn into_json(self, id: Option<&RawValue>) -> anyhow::Result<String>;
```
Comment (dvdplm) [inline types/src/v2/traits.rs:14]: There are no impls of this trait anywhere, which is a bit odd. Do we need this trait @niklasad1 ?

Comment (niklasad1) [reply types/src/v2/traits.rs:14]: No, I removed it in another pending PR.

Just a nit when moving those types from `ws server`

File: http-server/src/access_control/cors.rs [hunk 2]
```diff
@@ -49,22 +49,22 @@ pub enum OriginProtocol {
 pub struct Origin {
 	protocol: OriginProtocol,
 	host: Host,
-	as_string: String,
+	host_with_proto: String,
```
Comment (niklasad1) [inline http-server/src/access_control/cors.rs:52]: almost like we should refactor this to a separate type ^^ :)

I think this entire `access_control module` needs to checked a little bit more carefully, I don't remember but this matcher stuff could maybe be replaced by the `url crate` where they have functions for parsing url related stuff.

**Review by niklasad1 — APPROVED** [review]
:+1:

**Pull request merged**